src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,1845 @@
+            yield return getData("static object F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16) => null;", "F", "F", "System.Func<System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Int32, System.Object, System.Object>");
+            yield return getData("static object F(int _1, object _2, int _3, object _4, int _5, object _6, int _7, object _8, int _9, object _10, int _11, object _12, int _13, object _14, int _15, object _16, int _17) => null;", "F", "F", null);
+
+            object?[] getData(string methodDeclaration, string methodGroupExpression, string methodGroupOnly, string? expectedType) =>
+                new object?[] { methodDeclaration, methodGroupExpression, expectedType is null ? getExpectedDiagnostics(methodGroupExpression, methodGroupOnly) : null, expectedType };
+        }
+
+        public static IEnumerable<object?[]> GetMethodGroupImplicitConversionData()
+        {
+            return GetMethodGroupData((methodGroupExpression, methodGroupOnly) =>
+                {
+                    int offset = methodGroupExpression.Length - methodGroupOnly.Length;
+                    return new[]
+                        {
+                            // (6,29): error CS8915: The delegate type could not be inferred.
+                            //         System.Delegate d = F;
+                            Diagnostic(ErrorCode.ERR_CannotInferDelegateType, methodGroupOnly).WithLocation(6, 29 + offset)
+                        };
+                });
+        }
+
+        [Theory]
+        [MemberData(nameof(GetMethodGroupImplicitConversionData))]
+        public void MethodGroup_ImplicitConversion(string methodDeclaration, string methodGroupExpression, DiagnosticDescription[]? expectedDiagnostics, string? expectedType)
+        {
+            var source =
+$@"class Program
+{{
+    {methodDeclaration}
+    static void Main()
+    {{
+        System.Delegate d = {methodGroupExpression};
+        System.Console.Write(d.GetDelegateTypeName());
+    }}
+}}";
+            var comp = CreateCompilation(new[] { source, s_utils }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            if (expectedDiagnostics is null)
+            {
+                CompileAndVerify(comp, expectedOutput: expectedType);
+            }
+            else
+            {
+                comp.VerifyDiagnostics(expectedDiagnostics);
+            }
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var expr = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().Single().Initializer!.Value;
+            var typeInfo = model.GetTypeInfo(expr);
+            Assert.Null(typeInfo.Type);
```

Comment:
I would not have expected this to be null, I would have expected this to be the natural type, and be converted to `Delegate`.

Reply:
@333fred That was my expectation also. We discussed offline a bit with Chuck and Aleksey (who though `null` is fine), but we'll discuss more and see what VB does for precedent.
I'm fine with a follow-up issue.

---
In reply to: [618737229](https://github.com/dotnet/roslyn/pull/52448#discussion_r618737229) [](ancestors = 618737229)

Reply:
One more thought on `.Type`. It may be necessary to provide the `Action`/`Func` type in some of our APIs for use by the ASP.NET code generator (needs to generate the right signature). Let's check with ASP.NET folks

---
In reply to: [618831702](https://github.com/dotnet/roslyn/pull/52448#discussion_r618831702) [](ancestors = 618831702,618737229)

Reply:
@cston can you take a note to follow up on this, or add a prototype comment?

---
In reply to: [619275948](https://github.com/dotnet/roslyn/pull/52448#discussion_r619275948) [](ancestors = 619275948,618831702,618737229)

Reply:
The ASP.NET prototype relied on `GetSymbolInfo` to obtain the signature of the lambda ([here](https://github.com/davidfowl/uController/blob/aa8bcb4b30764e42bd72d326cb2718a4d4eaf4a9/src/uController.SourceGenerator/uControllerGenerator.cs#L62-L81)) so we should be good. Still worth following up on (I added a note to test plan).

Reply:
We should add some validation for `GetSymbolInfo` though.

Reply:
Logged [https://github.com/dotnet/roslyn/issues/52874](https://github.com/dotnet/roslyn/issues/52874).